modules/tinymce/src/core/main/ts/api/util/URI.ts

```diff
@@ -133,7 +133,7 @@ class URI {
     // Parse URL (Credits goes to Steave, http://blog.stevenlevithan.com/archives/parseuri)
     url = url.replace(/@@/g, '(mce_at)'); // Zope 3 workaround, they use @@something
 
-    const urlMatch = /^(?:(?![^:@]+:[^:@\/]*@)([^:\/?#.]+):)?(?:\/\/)?((?:(([^:@\/]*):?([^:@\/]*))?@)?([^:\/?#]*)(?::(\d*))?)(((\/(?:[^?#](?![^?#\/]*\.[^?#\/.]+(?:[?#]|$)))*\/?)?([^?#\/]*))(?:\?([^#]*))?(?:#(.*))?)/.exec(url);
+    const urlMatch = /^(?:(?![^:@]+:[^:@\/]*@)([^:\/?#.]+):)?(?:\/\/)?((?:(([^:@\/]*):?([^:@\/]*))?@)?(\[.*?\]|[^:\/?#]*)(?::(\d*))?)(((\/(?:[^?#](?![^?#\/]*\.[^?#\/.]+(?:[?#]|$)))*\/?)?([^?#\/]*))(?:\?([^#]*))?(?:#(.*))?)/.exec(url);
```

Comment:
This regex is very permissive. Could you please make it more specific, e.g. to only match on `[a-fA-F0-9:.]+` or similar?

Reply:
Is there any harm in being very permissive? Because URIs are not validated at all; only valid URIs are expected by the API. So it shouldn't really matter what regex we choose as long as it matches valid URIs correctly.

Reply:
As a general rule, we try to restrict any regex we use for performance, security and accuracy reasons. Your current solution would be slower than restricting the regex to only valid characters (a quick test on http://regex101.com/ suggests it may require 10 times as many steps), and has a much higher potential for incorrectly detecting other content. It also has a higher chance of causing a ReDoS.

Reply:
I see. Then `[a-zA-Z0-9:.%]+` is the simplest restrictive regex. We need `a-zA-Z` and `%` because of link-local IPv6 addresses with zone index.